### PR TITLE
More robust removal of temp files, renaming of output data to PAO1 and PA14 gene numbers & update of salmon params

### PR DESCRIPTION
**Pull request by georgiadoing**
## What

This request includes commits in which I 1) updated the shell and pbs scripts that run the salmon download and processing to make sure temp files (fastq) are deleted from local scratch even when the salmon call fails and 2) adds annotation files and scripts that take the salmon output and combine the quant.fs files from multiple samples into a compendium with PAO1 or PA14 gene numbers instead of transcriptome index assembly IDs and 3) updated salmon parameters.

## Why

The first commit was because I was getting memory-based call failures when local scratch was filling up after failed salmon calls.

The second is for convenience of exporting results in a way that makes them easy to browse since the PAO1 and PA14 numbers are more interpretable (to me at least) than the assembly IDs.

I changed the salmon parameters because the PAO1 samples from the wt vs anr experiment (https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE68534) had very low mapping rates (~2%) across all samples (WT and anr mutants).While the new parameters only increased the mapping rate marginally, they seem more logical to keep for all samples (see descriptions below).

## How

I added if statements to the pbs scripts that should catch a filed call to the bash script that runs the salmon call and proceed to delete any files in the temp dir instead of having the files in the temp dir be deleted in the bash script that calls salmon where they were not run if the salmon call failed. I will confirm with research computing that this is the right strategy to take.

The re-naming is done in R, using a simple mapping files (in the shape_comp/ dir) similarly to how I swap between PAO1 and PA14 numbers in R.

For the salmon params i first want to mention the change in the salmon index call (https://github.com/georgiadoing/pa-seq-compendia/blob/main/pbs_scripts/salmon_ti.pbs) -
**--k  15** changes the size of the mapping window from the default of 31 to 15 which can make the mapping more sensitive.

The other changes are in the salmon quant call (https://github.com/georgiadoing/pa-seq-compendia/blob/main/shell_scripts/salPa_byx.sh) -
**--softclip** allows the basepairs on the ends of fragments to mis-pair increasing sensitivity
**--softclipOverhangs** allows basepairs on the ends of reads to 'hang-off' or not map to any of the read, also increasing sensitivity
**--minScoreFraction 0.65** lowers the acceptable mapping score from 0.87 to 0.65 allowing for low-quality mappings to be kept, also increasing sensitivity
**--fldMean 51** sets the expected mean read length to 51 instead of 75 which fits with the reads from the PAO1 samples (although I have not checked read length for all samples in the compendium but expecting a shorter read increases the liklehood a local mapping will be kept)
**--rangeFactorizationBias 4** allows there to be 4 nt gaps between mapping windows
and
**--seqBias** corrects for sequence-specific biases that can make easily mapped reads through-off the expectations for less-well mapped reads, again potentially increasing sensitivity

Overall these changes are meant to increase sensitivity but have only really been used for 2 datasets (one PAO1 and one PA14). I still think they are good places to start and a more thorough test of the parameters on a larger subset of samples can be done at some point, though I do not expect it to influence mapping rates or transcriptional profiles too much, it still seems best practice.

### CODE REVIEW

File: pbs_scripts/salmon_ti.pbs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#PBS -e oe
+#PBS -o oe
+
+
+
+cd $PBS_O_WORKDIR
+
+module load salmon
+
+salmon index -t ../pao1_cdna.fa.gz -i ./t_ind/pao1_cdna_k15 -k 15
```
Comment (ajlee21) [inline pbs_scripts/salmon_ti.pbs:29]: Just wanted to confirm that we're using the same indices to help track down what might be happening with the mapping rates. 

I downloaded the GENE DNA file for Pseudomonas aeruginosa PAO1 (Reference) and Pseudomonas aeruginosa UCBPP-PA14 from http://www.pseudomonas.com/strain/download

`Pseudomonas_aeruginosa_PAO1_107.ffn.gz`
`Pseudomonas_aeruginosa_UCBPP-PA14_109.ffn.gz`

File: pbs_scripts/sra_to_salmon.pbs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#PBS -e /dartfs-hpc/scratch/f002bx6/salmon/sra_comp/oe
+#PBS -o /dartfs-hpc/scratch/f002bx6/salmon/sra_comp/oe
+
+
+# This is a job array submission script that passes array
+# indices and a directory to a bash script which initiates
+# the processing of the experiment-specific file (SRX or ERX)
+# in the specified directory at the specified index.
+
+# This level of parallelization, at the expeirment level,
```
Comment (ajlee21) [inline pbs_scripts/sra_to_salmon.pbs:29]: Nice description. Definitely very helpful for users!

File: pbs_scripts/sra_to_salmon_pa14.pbs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+
+cd $PBS_O_WORKDIR
+
+less $PBS_NODEFILE
+#echo $PBS_JOBID
+mkdir /scratch/$PBS_JOBID
+#ls -R /scratch/$PBS_JOBID
+
+if ! ../shell_scripts/sra2sal_byx.sh ${PBS_ARRAYID} pa14_k15 ; then
```
Comment (ajlee21) [inline pbs_scripts/sra_to_salmon_pa14.pbs:55]: Is the only difference between `sra_to_salmon.pbs ` and `sra_to_salmon_pa14.pbs` and `sra_to_salmon_pao1.pbs` that you have print statements to check on the jobs? Is `sra_to_salmon.pbs ` needed?

File: shape_comp/quant_collect.R [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+#print(sf_files)
+## parse file paths to get experiment accession
+sf_names <- sapply(list.files(args[1], pattern= '.sf', 
+                              recursive = TRUE, full.names = FALSE), function(x)
+                                substring(x, regexpr('/[E,S]RX',x)+1, regexpr('/quant',x)-1))
+# Read in the data from all quant files
+sf_datasets <- lapply(sf_files, function(x) read.csv(x, sep = '\t',
+                                                     stringsAsFactors = FALSE))
+# combine into a dataframes of read numbers and TPM, merging by name
+sf_all_reads <- Reduce(function(df1, df2) merge(df1, df2[,c(1,5)], by = 'Name'), sf_datasets)
```
Comment (ajlee21) [inline shape_comp/quant_collect.R:47]: Is there a reason you have a truncated dataframe and a full one?

**Review by ajlee21 — APPROVED** [review]
The code looks good to me. I'm not seeing an obvious reason for the mapping rate issue except for perhaps a difference in what reference we used, but I believe we're using the same file. 

I was mainly getting a low mapping when I combined the PAO1 with phage genes. But even then my mapping rates were around 30% not 2% so I'm not sure what is happening here. Maybe looking into the samples you downloaded to make sure you grabbed what you expected..?

File: pbs_scripts/salmon_ti.pbs (hunk 1 above)
Comment (georgiadoing) [reply pbs_scripts/salmon_ti.pbs:29]: I downloaded the cDNA references from ensembleBacteria ([https://github.com/hoganlab-dartmouth/pa-seq-compendia/blob/main/t_indxs/ensemble_urls.txt](url))  . I think they  should be pretty similar to the gene-based ones from either ensemble or from your links above, but you are right that we should check. 

The links to ensemble that aren't ftp just for reference:
( [https://bacteria.ensembl.org/Pseudomonas_aeruginosa_pao1_gca_000006765/Info/Index](url)
[https://bacteria.ensembl.org/Pseudomonas_aeruginosa_ucbpp_pa14_gca_000014625/Info/Index](url) )

**Pull request merged**